server/src/main/java/com/linecorp/centraldogma/server/internal/replication/ZooKeeperCommandExecutor.java

```diff
@@ -1072,7 +1070,23 @@ private static String pathFromRevision(long revision) {
             }
 
             final T result = delegate.execute(command).get();
-            final ReplicationLog<T> log = new ReplicationLog<>(replicaId(), command, result);
+            final ReplicationLog<?> log;
+            if (command.type() == CommandType.NORMALIZING_PUSH) {
+                final NormalizingPushCommand normalizingPushCommand = (NormalizingPushCommand) command;
+                assert result instanceof CommitResult : result;
+                final CommitResult commitResult = (CommitResult) result;
+                final Revision commitResultRevision = commitResult.revision();
+                final Command<Revision> pushAsIsCommand = Command.pushAsIs(
+                        normalizingPushCommand.timestamp(), normalizingPushCommand.author(),
+                        normalizingPushCommand.projectName(), normalizingPushCommand.repositoryName(),
+                        commitResultRevision.backward(1),
+                        normalizingPushCommand.summary(), normalizingPushCommand.detail(),
+                        normalizingPushCommand.markup(),
+                        commitResult.changes());
+                log = new ReplicationLog<>(replicaId(), pushAsIsCommand, commitResultRevision);
+            } else {
+                log = new ReplicationLog<>(replicaId(), command, result);
+            }
```

Comment:
- Could it be convenient if we add `Command<Revision> pushAsIsCommand = normalizingPushCommand.asIs(commitResult)`?
- Add `assert !(command instanceof NormalizingPushCommand)` to the constructor of `ReplicationLog`?